src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserAlreadyExistException;
+import core.basesyntax.exception.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_PASSWORD_LENGTH = 6;
+    private static final int VALID_LOGIN_LENGTH = 6;
+    private static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        isUserNull(user);
+        isValidUserLogin(user);
+        isValidUserPassword(user);
+        isValidUserAge(user);
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserAlreadyExistException("User: " + user + ", already exist in storage");
+        }
```

Comment:
let's make your code consistent and move this piece of code to a separate method. I'd recommend putting it in your 'isValidUserLogin' method